### PR TITLE
Working_logic_for_comments

### PR DESCRIPTION
**Pull request by Andrii-Nechiporuk**

### CODE REVIEW

File: Server/IT-Community.Server.Core/Entities/Comment.cs [hunk 1]
```diff
@@ -13,6 +13,7 @@ public class Comment
         [Key]
         [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
         public int Id { get; set; }
+        public int? ParentId { get; set; }
```
Comment (vladtymo) [inline Server/IT-Community.Server.Core/Entities/Comment.cs:17]: Add navigation property for this relation.

File: Server/IT-Community.Server/Controllers/CommentController.cs [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        [Authorize]
+        public async Task<IActionResult> DeletePost(int id)
+        {
+            var userId = GetUserId();
+            await _commentService.DeleteComment(id, userId);
+            return Ok();
+        }
+
+        [NonAction]
+        public string GetUserId()
```
Comment (vladtymo) [inline Server/IT-Community.Server/Controllers/CommentController.cs]: Use service method instead

File: Server/IT-Community.Server.Core/Entities/Comment.cs (hunk 1 above)
Comment (vladtymo) [inline Server/IT-Community.Server.Core/Entities/Comment.cs:17]: Also you should add collection navigation property for replies.

File: Server/IT-Community.Server.Infrastructure/Services/CommentService.cs [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+            _mapper = mapper;
+            _userManager = userManager;
+        }
+
+        public List<CommentPostDto> GetCommentsWithReplies(int postId)
+        {
+            List<CommentPostDto> mainResult = new();
+            List<CommentPostDto> subtList = new();
+
+            var commentList = _mapper.Map<List<CommentPostDto>>(_unitOfWork.CommentRepository.GetAll(x => x.PostId == postId && x.ParentId == null).ToList());
```
Comment (vladtymo) [inline Server/IT-Community.Server.Infrastructure/Services/CommentService.cs]: Use navigation properties to load replies.

File: Server/IT-Community.Server.Infrastructure/Services/CommentService.cs [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+            var allList = _mapper.Map<List<CommentPostDto>>(_unitOfWork.CommentRepository.GetAll(x => x.PostId == postId).ToList());
+
+            foreach (var item in commentList)
+            {
+                subtList = commentList.Where(x => x.Id == item.Id).Select(x => new CommentPostDto()
```
Comment (vladtymo) [inline Server/IT-Community.Server.Infrastructure/Services/CommentService.cs]: Use auto mapper methods to map it.

**Review by vladtymo — CHANGES_REQUESTED** [review]

**Review by basilioss — APPROVED** [review]

File: Server/IT-Community.Server.Core/DataContext.cs [hunk 5]
```diff
@@ -27,6 +27,7 @@ protected override void OnModelCreating(ModelBuilder builder)
             builder.Entity<Post>().Navigation(e => e.Forum).AutoInclude();
             builder.Entity<Comment>().Navigation(e => e.User).AutoInclude();
 
+
```
Comment (vladtymo) [inline Server/IT-Community.Server.Core/DataContext.cs]: Why you put double line break?

File: Server/IT-Community.Server/Program.cs [hunk 6]
```diff
@@ -84,7 +86,7 @@
 
 app.UseStaticFiles();
 
-app.UseMiddleware<ExceptionMiddleware>();
+//app.UseMiddleware<ExceptionMiddleware>();
```
Comment (vladtymo) [inline Server/IT-Community.Server/Program.cs]: Why commented?

File: Server/IT-Community.Server.Infrastructure/Services/CommentService.cs [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+            }
+        }
+
+        private List<CommentPostDto> GetAllReplies(ref List<CommentPostDto> comments, int parentId)
+        {
+            List<CommentPostDto> innerComments = comments;
+
+            return comments.Where(x => x.ParentId == parentId).Select(m => new CommentPostDto()
+            {
+                Id = m.Id,
```
Comment (vladtymo) [inline Server/IT-Community.Server.Infrastructure/Services/CommentService.cs]: Use auto mapper instead these lines.

File: Server/IT-Community.Server.Infrastructure/Services/CommentService.cs [hunk 8]
```diff
@@ -0,0 +1,153 @@
...
+            _userManager = userManager;
+        }
+
+        public List<CommentPostDto> GetCommentsWithReplies(int postId)
+        {
+            var commentList = _mapper.Map<List<CommentPostDto>>
+                (_unitOfWork.CommentRepository.GetAll(x => x.PostId == postId && x.ParentId == null, includeProperties: new[] {"Comments"}));
+            foreach(var comment in commentList)
+            {
+                LoadComments(comment, postId);
```
Comment (vladtymo) [inline Server/IT-Community.Server.Infrastructure/Services/CommentService.cs:38]: But the replies should be already loaded by using Include()... or not? I think, you can avoid this loading cycle.

File: Server/IT-Community.Server.Core/DataContext.cs (hunk 5 above)
Comment (Andrii-Nechiporuk) [reply Server/IT-Community.Server.Core/DataContext.cs]: deleted

File: Server/IT-Community.Server/Program.cs (hunk 6 above)
Comment (Andrii-Nechiporuk) [reply Server/IT-Community.Server/Program.cs]: > Why commented?

forgot to uncomment

File: Server/IT-Community.Server.Infrastructure/Services/CommentService.cs (hunk 7 above)
Comment (Andrii-Nechiporuk) [reply Server/IT-Community.Server.Infrastructure/Services/CommentService.cs]: This method i don`t use, but forgot to delete

File: Server/IT-Community.Server.Infrastructure/Services/CommentService.cs (hunk 8 above)
Comment (Andrii-Nechiporuk) [reply Server/IT-Community.Server.Infrastructure/Services/CommentService.cs:38]: They`re not loading automaticatilly, only first layer. BUt there can be much more, so we need to load them in this method

**Review by vladtymo — APPROVED** [review]

**Pull request merged**